ui/plugin-system/src/runtime/TimeRangeProvider/TimeRangeProvider.tsx

```diff
@@ -56,37 +57,104 @@ export function useTimeRange(): TimeRange {
   return useTimeRangeContext();
 }
 
+/**
+ * Build the time range context from some initial state.
+ * The values inside the initial state can be different depending on if the query params has been enabled and/or if we
+ * gave some initial time range / refresh interval.
+ * @param ctx
+ */
+function useContextMemo(
+  ctx: Omit<TimeRange, 'refreshKey' | 'absoluteTimeRange' | 'refresh' | 'refreshIntervalInMs'>
+): TimeRange {
+  const [refreshCounter, setRefreshCounter] = useState(0);
+
+  const refresh = useCallback(() => {
+    setRefreshCounter(refreshCounter + 1);
```

Comment:
```suggestion
    setRefreshCounter(prevCounter => prevCounter + 1);
```
An you can remove the useCallback wrapping the function

Reply:
You mean something like this ? 
![image](https://github.com/perses/perses/assets/10258608/789fdc7c-1ce1-401f-a993-8c106c11e054)


My IDE is not happy with that.
![image](https://github.com/perses/perses/assets/10258608/8ba89b00-8809-464f-ba32-ae296023c39f)



TBH I don't want to modify too much the existing because I'm afraid of regression

Reply:
Ah yes, never mind about the useCallback removal